### DISCUSSION TITLE
this.bclient.options.connectionName = connectionName;

### DISCUSSION THREAD

**Discussion by jens-dehnert-winkler-ay** [OptimalBits/bull#2762] [Q&A]
In worker.js the worker redis connection name is replaced with a base64 encoded string of the queuename. 

Is there any reason why this is done?

For some background around the question: our instrumentation plugins gather data based on the connectionName. 
As the connection name can be provided as a parameter, when setting up the Bull instance with a redis client, it seems very obscure that its replaced, and I wonder if its really necessary.

Cheers!

**Comment by manast** [comment]
I think this was done to normalise the characters used by the connection name so that they could more easy be parsed when listing all the workers associated to a given queue.
